src/neuroconv/tools/roiextractors/imagingextractordatachunkiterator.py

```diff
@@ -97,19 +96,25 @@ def _get_scaled_buffer_shape(self, buffer_gb: float, chunk_shape: tuple) -> tupl
                 for dimension_index, dimension_length in enumerate(max_buffer_shape)
             ]
         )
-
         return scaled_buffer_shape
 
     def _get_dtype(self) -> np.dtype:
         return self.imaging_extractor.get_dtype()
 
     def _get_maxshape(self) -> tuple:
-        return (self.imaging_extractor.get_num_frames(),) + self.imaging_extractor.get_image_size()[::-1]
+        video_shape = (self.imaging_extractor.get_num_frames(),)
+        image_shape = self.imaging_extractor.get_image_size()
+        width, height = image_shape[1], image_shape[0]  # ROIExtractors convention is flipped
+        video_shape += (width, height)
+        if len(image_shape) == 3:
+            depth = image_shape[2]
+            video_shape += (depth,)
+        return video_shape
 
     def _get_data(self, selection: Tuple[slice]) -> np.ndarray:
         data = self.imaging_extractor.get_video(
             start_frame=selection[0].start,
             end_frame=selection[0].stop,
         )
-
-        return data.transpose((0, 2, 1))[(slice(0, self.buffer_shape[0]),) + selection[1:]]
+        tranpose_axes = (0, 2, 1) if len(data.shape) == 3 else (0, 2, 1, 3)
+        return data.transpose(tranpose_axes)[(slice(0, self.buffer_shape[0]),) + selection[1:]]
```

Comment:
Out of curiosity, why the `selection[1:]` here?

Reply:
`selection[0]` is the main slice through `get_video` until ROIExtractors supports efficient image slicing (if it even can for certain extractors... safer to just assume frame-wise access)

Then the remaining axes of selection could be 2 or 3 (possibly 5 for sbx?) dimensional, but that's handled automatically by the parent class.

Reply:
Thanks.